web/src/containers/Main/hooks/index.ts

```diff
@@ -55,20 +64,63 @@ export function useMain(questionnaireId: string) {
         [questionnaireId, fhirMode],
     );
 
+    const showModal = (response: any, type: string) => {
+        if (type === 'success') {
+            toast.success('New mapper created');
+        } else {
+            toast.error(
+                formatError(response.error, {
+                    mapping: { conflict: 'Please reload page' },
+                    format: (errorCode, errorDescription) =>
+                        `An error occurred: ${errorDescription} (${errorCode}). Please reach tech support`,
+                }),
+            );
+        }
+    };
+
     const saveQuestionnaireFHIR = useCallback(
         async (resource: Questionnaire) => {
-            const response = await service({
+            const response = await service<unknown, OperationOutcome>({
                 method: 'PUT',
                 data: resource,
                 url: `/${fhirMode ? 'fhir/' : ''}Questionnaire/${resource.id}`,
             });
+
             if (isSuccess(response)) {
                 questionnaireManager.reload();
             } else {
-                console.error('Could not save Questionnaire:', response.error.toString());
+                if (
```

Comment:
Error checking and id extraction could be placed to a function.
When it is done, you can cover the function with unit tests.